app/src/main/java/com/example/myapplication/ui/home/recycler/PostedImageViewHolder.kt

```diff
@@ -1,59 +1,100 @@
 package com.example.myapplication.ui.home.recycler
 
 import android.view.View
+import android.widget.Toast
 import androidx.recyclerview.widget.RecyclerView
 import com.bumptech.glide.Glide
 import com.example.myapplication.R
 import com.example.myapplication.ui.home.model.PostedItem
+import com.example.myapplication.ui.home.model.PostedItemActionType
+import com.example.myapplication.utils.DoubleClickListener
 import kotlinx.android.extensions.LayoutContainer
 import kotlinx.android.synthetic.main.posted_item.view.*
 
 class PostedImageViewHolder(
     override val containerView: View,
-    private val action: ((String) -> Unit)?
+    private val action: ((PostedItemActionType) -> Unit)?
 ) : RecyclerView.ViewHolder(containerView),
     LayoutContainer {
 
     companion object {
         const val LAYOUT_ID = R.layout.posted_item
     }
 
+    var isLiked: Boolean = false
+    var likesCount: Int = 0
+
     fun bind(model: PostedItem) {
-        containerView.tvTopProfileName.text = model.profile.profileName
+
+        isLiked = model.isLiked
+        likesCount = if (isLiked) model.likesCount - 1 else model.likesCount + 1
+        setLike(isLiked)
+
+        containerView.ivTestCommentBottom.setOnClickListener {
+            action?.invoke(
+                PostedItemActionType.Comment(
+                    model.postId,
+                    containerView.etAddComment.text.toString(),
+                    System.currentTimeMillis().toString()
+                ) as PostedItemActionType
+            )
+        }
+
+        containerView.tvTopProfileName.text = model.profile.name
         containerView.tvTopProfileName.setOnClickListener {
-            action?.invoke(model.profile.profileLink)
+            action?.invoke(PostedItemActionType.ProfileId(model.profile.id) as PostedItemActionType)
         }
 
         Glide.with(containerView.context)
-            .load(model.profile.profilePhoto)
+            .load(model.profile.photo)
             .into(containerView.civProfilePhoto)
+
         containerView.civProfilePhoto.setOnClickListener {
-            action?.invoke(model.profile.profileLink)
+            action?.invoke(PostedItemActionType.ProfileId(model.profile.id) as PostedItemActionType)
         }
 
         Glide.with(containerView.context)
             .load(model.imageUri)
             .into(containerView.ivPostPhoto)
 
-//        containerView.ivIsLiked.
-//        containerView.etAddComment.setOnClickListener {
-//            action?.
-//        }
-
-//        containerView.ivShare.setOnClickListener {
-//            action?.
-//        }
-
-//        containerView.ivBottomIsLiked
+        containerView.ivPostPhoto.setOnClickListener(object : DoubleClickListener() {
+            override fun onDoubleClick(v: View) {
```

Comment:
Dlaczego obsługa like jest na double click? :) To raczej niespotykane w świecie mobile.